src/components/AvatarGroup/avatars.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import StyledAvatar from './styled/avatar';
+
+export default function Avatars(props) {
+    const { avatars, maxAvatars, showCounter, size } = props;
+    const total = avatars.length;
+    const maxAvatar = total > maxAvatars && showCounter ? maxAvatars - 1 : maxAvatars;
+
+    const items = avatars.slice(0, maxAvatar).map((avatar, idx) => {
+        const { src, assistiveText, title, initials, icon, initialsVariant } = avatar;
+        const key = `avatar-${idx}`;
+        const zIndex = `${maxAvatar - idx}`;
+        const iconImg = <img src={icon} alt="icon" />;
```

Comment:
As defined in the interface the icon prop is a node so you should pass it directly to Avatar.